### DISCUSSION TITLE
Coupons not being enabled

### DISCUSSION THREAD

**Discussion by mbpassos** [duncanmcclean/simple-commerce#1097] [Help]
- php 8.3
- Laravel 11
- Statamic v5
- Fresh Simple Commerce Starter Kit install (products, entries, etc in general nothing has not been modified from the dedault configs of the starter kit).

I tried to set up a simple test coupon but it does not get enabled, even qhen I set the expire date:

This is the coupon's .yaml file:
```
description: test
customer_eligibility: all
maximum_uses: 10
expires_at: '2024-06-30'
id: 15526646-9336-4dae-b02b-236f31679c16
code: TEST
value: 100
type: fixed
```

It does not show up in the view `/checkout/information`

When I add the tag `{{ sc:coupon:code }}` in a simple div in the information view in order to see if any coupon information shows up I get the following error:
`property_exists(): Argument #1 ($object_or_class) must be of type object|string, null given`

Also, when I click on Save after creating the coupon I get the allert window as if I was trying to leave the page withou saving. I then hit Cancel and the Saved Toast shows up afer that. So something weird is going on.

I installed the addon in an xhisting project and I get the same exact problems with the coupons.

**Comment by duncanmcclean** [comment]
> When I add the tag {{ sc:coupon:code }} in a simple div in the information view in order to see if any coupon information shows up I get the following error:
> `property_exists(): Argument #1 ($object_or_class) must be of type object|string, null given`

Can you wrap the `sc:coupon:code` stuff inside an if statement to check if a coupon has been redeemed:

```
{{ if {sc:coupon:has} === true }}
    {{ sc:coupon:code }}
{{ /if }}
```

Here's an example of how the redeeming/displaying of coupons is handled in the Starter Kit:

https://github.com/duncanmcclean/sc-starter-kit/blob/a23b38106d9d09f53866ea992e19069e2c411a2a/resources/views/checkout/_cart.antlers.html#L48-L91

**Reply by mbpassos** [reply]
Doing like that remover the error message, but it does not show the info on the coupon that has been created:

```
{{ if {sc:coupon:has} === true }}
    {{ sc:coupon:code }}
{{ /if }}
```

**Reply by duncanmcclean** [reply]
Hmm, you *will* need to have that check around it, otherwise it'll error. However, it should be displaying the coupon code.

Are you able to reproduce the issue on a fresh site or provide access to your project so I can take a look? Thanks.

**Reply by mbpassos** [reply]
Thank you, Duncan. This is the copy of the project https://github.com/mbpassos/e-commerce

**Reply by duncanmcclean** [reply]
Thanks! I'll take a look when I'm able.

**Reply by duncanmcclean** [reply]
I've just cloned down your project and I'm not experiencing any issues with coupons.

I copied the example coupon you originally provided into `content/simple-commerce/coupons` and redeemed it on the checkout payment page, without any errors:

![CleanShot 2024-06-21 at 10 41 07](https://github.com/duncanmcclean/simple-commerce/assets/19637309/eb2b28de-3fd3-413e-9a97-046de3b68d7a)

I did have to comment out [these](https://github.com/mbpassos/e-commerce/blob/2993794d94eda2e9e363a338f25326e796d7f65f/resources/views/checkout/_cart.antlers.html#L52) [lines](https://github.com/mbpassos/e-commerce/blame/2993794d94eda2e9e363a338f25326e796d7f65f/resources/views/checkout/_cart.antlers.html#L97) to get the "redeem coupon" form to show, but not because it was erroring.
